plugins/common/tls/config.go

```diff
@@ -75,6 +77,11 @@ func (c *ClientConfig) TLSConfig() (*tls.Config, error) {
 		tlsConfig.RootCAs = pool
 	}
 
+	if c.TLSCertAndKey != "" {
+		c.TLSCert = c.TLSCertAndKey
+		c.TLSKey = c.TLSCertAndKey
+	}
```

Comment:
Can we use a combined cert+key without adding a new setting by setting TLSCert and TLSKey to the same filename? If so, why do we need a new setting?

Reply:
I'm mirroring how MongoDB expects its tls pem files. MongoDB anticipates the key and certificate to be combined in a single file and therefore do not require 2 separate options/settings like telegrafs `tls_key` and `tls_cert` options. `tls_cert_key` would be akin to [MongoDB's --tlsCertificateKeyFile] option. (https://docs.mongodb.com/mongodb-shell/reference/options/#std-option-mongosh.--tlsCertificateKeyFile). I could also ask why anyone would not want the added flexibility?

1. It's more telling about the content of the provided file. (tls_key only has the key, tls_cert only has the cert, tls_cert_key has both)
2. The telegraf configuration looks cleaner.
2. Administrators don't have to tailor their pem files to conform to telegrafs individual tls_cert and tls_key files options.

Reply:
My concern about adding a new setting is that people will think they need to use it. We get lots of people asking for help with this kind of stuff on slack and community messageboards. Having a simple configuration with minimal settings seems to help people configure telegraf without needing as much help.

I think we should stick with two settings, cert and key, then put a note in the documentation and sample config that if you have a combined file that has both cert and key, to put the filename in both settings.

Reply:
@bustedware see this comment about adding docs for using a combined cert+key file.

Reply:
Hi,

I don't think deprecating those existing options is a wise choice for our users. This would be both disruptive to existing users. It is also completely valid for these to be two different files.

Having config options that do the same thing, but are slightly different is actually far less simple for users as it leads to confusion as to which one to use, and when. I would also say we do not have any bugs where users are confused as-is, however, I am willing to think this would change when we add another config option.

The addition of the `tls_key_password` is a great thing for our users, however, this PR should probably stick to just that for now and document that the same file can be used for both the key and cert if required.

Thanks

Reply:
Hey @powersj closing this PR since `tls_key_pwd` already exists in master. Best regards

Reply:
Hey @bustedware, could you point me to the PR where that’s added?

Reply:
Found it here https://github.com/influxdata/telegraf/pull/9923

Reply:
You "found it"? It's a PR also created by yourself. I'm confused now.

Also, that one doesn't have extra checks and code in `common/tls/config.go` as this one has.